src/Discord.Net.Rest/Entities/Messages/EmbedBuilder.cs

```diff
@@ -8,14 +8,36 @@ public class EmbedBuilder
     {
         private readonly Embed _embed;
 
+        public const int MaxFieldCount = 25;
+        public const int MaxTitleLength = 256;
+        public const int MaxDescriptionLength = 2048;
+
         public EmbedBuilder()
         {
             _embed = new Embed("rich");
             Fields = new List<EmbedFieldBuilder>();
         }
 
-        public string Title { get { return _embed.Title; } set { _embed.Title = value; } }
-        public string Description { get { return _embed.Description; } set { _embed.Description = value; } }
+        public string Title
+        {
+            get => _embed.Title;
+            set
+            {
+                if (value?.Length > MaxTitleLength) throw new ArgumentException($"Title length must be less than or equal to {MaxTitleLength}.");
+                _embed.Title = value;
+            }
+        }
+
+        public string Description
+        {
+            get => _embed.Description;
+            set
+            {
+                if (value?.Length > MaxDescriptionLength) throw new ArgumentException($"Description length must be less than or equal to {MaxDescriptionLength}.");
+                _embed.Description = value;
+            }
+        }
+
         public string Url { get { return _embed.Url; } set { _embed.Url = value; } }
```

Comment:
I concur; using the Uri type means that we get validation of the input as an artefact of its implementation, and IMO the Uri class is nicer to work with.